ts/parts/Pointer.ts

```diff
@@ -2106,51 +2127,83 @@ class Pointer {
             ownerDoc = container.ownerDocument;
 
         container.onmousedown = function (e: MouseEvent): void {
-            pointer.onContainerMouseDown(e as any);
+            pointer.onContainerMouseDown(pointer.normalize(e));
         };
         container.onmousemove = function (e: MouseEvent): void {
-            pointer.onContainerMouseMove(e as any);
+            pointer.onContainerMouseMove(pointer.normalize(e));
         };
         container.onclick = function (e: MouseEvent): void {
-            pointer.onContainerClick(e as any);
+            pointer.onContainerClick(pointer.normalize(e));
         };
         this.unbindContainerMouseLeave = addEvent(
             container,
             'mouseleave',
-            pointer.onContainerMouseLeave as any
+            function (e: MouseEvent): void {
+                pointer.onContainerMouseLeave(pointer.normalize(e));
+            }
         );
         if (!H.unbindDocumentMouseUp) {
             H.unbindDocumentMouseUp = addEvent(
                 ownerDoc,
                 'mouseup',
-                pointer.onDocumentMouseUp as any
+                function (e: MouseEvent): void {
+                    pointer.onDocumentMouseUp(pointer.normalize(e));
+                }
             );
         }
         if (H.hasTouch) {
             addEvent(
                 container,
                 'touchstart',
                 function (e: TouchEvent): void {
-                    pointer.onContainerTouchStart(e as any);
+                    pointer.onContainerTouchStart(pointer.normalize(e));
                 }
             );
             addEvent(
                 container,
                 'touchmove',
                 function (e: TouchEvent): void {
-                    pointer.onContainerTouchMove(e as any);
+                    pointer.onContainerTouchMove(pointer.normalize(e));
                 }
             );
             if (!H.unbindDocumentTouchEnd) {
                 H.unbindDocumentTouchEnd = addEvent(
                     ownerDoc,
                     'touchend',
-                    pointer.onDocumentTouchEnd as any
+                    function (e: TouchEvent): void {
+                        pointer.onDocumentTouchEnd(pointer.normalize(e));
+                    }
                 );
             }
         }
     }
 
+    /**
+     * Sets the index of the hovered chart and leaves the previous hovered
+     * chart, to reset states like tooltip.
+     *
+     * @private
+     * @function Highcharts.Pointer#setHoverChartIndex
+     */
+    public setHoverChartIndex(): void {
+        const chart = this.chart;
+        const hoverChart = H.charts[H.hoverChartIndex || -1];
```

Comment:
Should be:
`const hoverChart = H.charts[pick(H.hoverChartIndex, -1)];`

... to prevent the 0 case failure.